dipy/segment/tests/test_feature.py

```diff
@@ -284,6 +285,26 @@ def extract(self, streamline):
     d2 = metric.dist(features1, features2)
     assert_equal(d1, d2)
 
+    class ArcLengthFeature(dipymetric.Feature):
+        def infer_shape(self, streamline):
+            if sys.version_info > (3,):
```

Comment:
Less ugly but still hard to understand.  Could y'all remind me why this is necessary?  Isn't the return type dealt with by https://github.com/nipy/dipy/pull/829/files#diff-6102332b6a9436b5c5b18ab9ee552a9eR169 ?